pymc/gp/cov.py

```diff
@@ -548,7 +568,48 @@ def full(self, X, Xs=None):
         return pt.exp(-0.5 * r)
 
 
-class ExpQuad(Stationary):
+class GeneralizedPeriodic(Stationary):
+    r"""
+    The Generalized Periodic kernel.
+
+    Can be used to wrap any isotropic stationary kernel to transform it
+    into a periodic version. The derivation can be achieved by mapping the
+    original inputs through the transformation u = (cos(x), sin(x)).
+    """
+
+    def __init__(self, base_kernel: IsotropicStationary, period):
```

Comment:
I would try to mimic the initialization signature that the `Periodic` kernel uses.

```suggestion
    def __init__(self, input_dim, base_kernel: IsotropicStationary, period, ls=None, ls_inv=None, active_dims=None):
```

I'm not sure if we want to keep the `ls`, `ls_inv`, etc or just use a `**kwargs` unpacking though.

By the way, is your idea to pass in an instance of an `IsotropicSationary` kernel, or the class? In the former case, you would need to make sure that the `input_dim`, and other parameters match with the `base_kernel`. In the latter case, you can construct the `base_kernel` inside of `__init__`, and make sure that you get consistent `active_dims` and `input_dim`.

Reply:
I like where you're going with this. If we went with this approach could we then simply "enhance" the existing `Periodic` kernel instead of creating a new one? e.g.

```python
def __init__(self, input_dim, period, ls=None, ls_inv=None, active_dims=None, base_kernel_class: Type[IsotropicKernel] = ExpQuad):
        super().__init__(input_dim, ls, ls_inv, active_dims)
        self.base_kernel_class = base_kernel_class
        self.period = period

@property
def base_kernel(self) -> IsotropicKernel:
    return self.base_kernel_class(self.input_dim, ls=self.ls, ls_inv=self.ls_inv, active_dims=self.active_dims)
```

...where the instance is instantiated on-demand so we're not storing `input_dim`, `ls`, `ls_inv` in two places?

Reply:
@lucianopaz - re-worked the change such that the original `Periodic` kernel is now extended in a backward compatible way. And the base kernel is passed as a class.